File: baseble/src/main/java/com/fcbox/ble/baseble/exception/BleException.java
package com.fcbox.ble.baseble.exception;


import com.fcbox.ble.baseble.common.BleExceptionCode;

import java.io.Serializable;


/**
 * BLE异常基类
 */
@SuppressWarnings("unused")
public class BleException implements Serializable {
    private BleExceptionCode code;
    private String description;

    public BleException(BleExceptionCode code, String description) {
        this.code = code;
        this.description = description;
    }

    public BleExceptionCode getCode() {
        return code;
    }

    public BleException setCode(BleExceptionCode code) {
        this.code = code;
        return this;
    }

    public String getDescription() {
        return description;
    }

    public BleException setDescription(String description) {
        this.description = description;
        return this;
    }

    @Override
    public String toString() {
        return "BleException{" +
                "code=" + code +
                ", description='" + description + '\'' +
                '}';
    }
}

File: baseble/src/main/java/com/xzy/ble/baseble/exception/BleException.kt
package com.xzy.ble.baseble.exception


import com.xzy.ble.baseble.common.BleExceptionCode

import java.io.Serializable


/**
 * BLE异常基类
 */
@Suppress("unused")
open class BleException(private var code: BleExceptionCode?, var description: String?) : Serializable {

    fun getCode(): BleExceptionCode? {
        return code
    }

    fun setCode(code: BleExceptionCode): BleException {
        this.code = code
        return this
    }

    fun getExDescription(): String? {
        return description
    }

    fun setDescription(description: String): BleException {
        this.description = description
        return this
    }

    override fun toString(): String {
        return "BleException{" +
                "code=" + code +
                ", description='" + description + '\''.toString() +
                '}'.toString()
    }
}
